db/seeds/roles.rb

```diff
@@ -0,0 +1,128 @@
+                          :edit_own_time_entries,
+                          :view_project_associations,
+                          :view_timelines,
+                          :edit_timelines,
+                          :delete_timelines,
+                          :view_reportings,
+                          :edit_reportings,
+                          :delete_reportings,
+                          :manage_wiki,
+                          :manage_wiki_menu,
+                          :rename_wiki_pages,
+                          :change_wiki_parent_page,
+                          :delete_wiki_pages,
+                          :view_wiki_pages,
+                          :export_wiki_pages,
+                          :view_wiki_edits,
+                          :edit_wiki_pages,
+                          :delete_wiki_pages_attachments,
+                          :protect_wiki_pages,
+                          :list_attachments,
+                          :add_messages,
+                          :edit_own_messages,
+                          :delete_own_messages,
+                          :browse_repository,
+                          :view_changesets,
+                          :commit_access,
+                          :view_commit_author_statistics]
+    end.save!
+
+    Role.new.tap do |reader|
+      reader.name = I18n.t(:default_role_reader)
+      reader.position =  5
+      reader.permissions = [:view_work_packages,
+                            :add_work_package_notes,
+                            :edit_own_work_package_notes,
+                            :save_queries,
+                            :view_calendar,
+                            :comment_news,
+                            :view_project_associations,
+                            :view_timelines,
+                            :view_reportings,
+                            :view_wiki_pages,
+                            :export_wiki_pages,
+                            :view_wiki_edits,
+                            :edit_wiki_pages,
+                            :list_attachments,
+                            :add_messages,
+                            :edit_own_messages,
+                            :delete_own_messages,
+                            :browse_repository,
+                            :view_changesets]
+    end.save!
+
+    Role.non_member.update_attributes name: I18n.t(:default_role_non_member),
+                                      permissions: [:view_work_packages,
+                                                    :view_calendar,
+                                                    :comment_news,
+                                                    :browse_repository,
+                                                    :view_changesets,
+                                                    :view_wiki_pages]
+
+    Role.anonymous.update_attributes name: I18n.t(:default_role_anonymous),
+                                     permissions: [:view_work_packages,
```

Comment:
@NobodysNightmare no it doesn't. The roles Anonymous and NonMember are not existing in non public projects.